### DISCUSSION TITLE
[Hibernate Reactive] Using with hibernate reactive to just build query strings?

### DISCUSSION THREAD

**Discussion by agreedSkiing** [Blazebit/blaze-persistence#1715] [Q&A] [answered]
Been reading the https://github.com/Blazebit/blaze-persistence/issues/1328 issue and started to wonder if it currently possible to just use Blaze to create query strings or normal criteria queries and use them in the Hibernate Reactive session factory? For me Blaze provides an easier way to create criteria queries with how the queries are build more like normal SQL, JPQL and HQL.

Or is the a similar Maven dependency out there that might help with this? Asking mostly because my current project needs to build dynamic queries which for us means optional parameters depending on what filters are used in our big `get All` rest endpoint.

Or can I pass the Mutiny the criteriabuilder to Blaze somehow?
![bild](https://user-images.githubusercontent.com/93997033/225435959-27238d90-7606-46ab-afc9-1fe961fcc6ea.png)

**Comment by beikov** [comment, marked as answer]
It is possible to do that, but note that there are limits. Certain features of Blaze-Persistence require deep integration which makes it very hard because Hibernate Reactive is so different. With Hibernate 6, the need for certain deep integrations became unnecessary, because it supports set operations and limit/offset natively.

You can try to use the query string from `CriteriaBuilder.getQueryString()` directly. You just need to bind through the parameters, but these are accessible through `getParameters()` and `getParameterValue()`
